### PR TITLE
ES5

### PR DESCRIPTION
**Pull request by solomonhawk**
Rebased old branch with master, updated so tests are passing.

### CODE REVIEW

**Comment by dce** [comment]
:+1: looks great

File: languages/JavaScript/src/decrypt.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+function decrypt(cipherText, key) {
+  // split the cipher text into pairs of characters
+  var cipherPairs = cipherText.match(/.{1,2}/g);
```
Comment (nhunzaker) [inline languages/JavaScript/src/decrypt.js]: Neat

File: languages/JavaScript/src/decrypt.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      // XOR the encoded char with the hex char, convert to characters
+      return strfchar(encodedChar ^ hexChar);
+    }).join('');
+}
+
+function strfchar(code) {
+  var cache = {};
+  return cache[code] || (cache[code] = String.fromCharCode(code));
+}
```
Comment (nhunzaker) [inline languages/JavaScript/src/decrypt.js]: Cool to see thoughts on memorization like this, do you know how much faster it makes things?

**Comment by nhunzaker** [comment]
:+1:

File: languages/JavaScript/Rakefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+namespace :JavaScript do
+  task :clean do
```
Comment (reagent) [inline languages/JavaScript/Rakefile]: This is optional -- this won't be run if it doesn't exist.

**Comment by dce** [comment]
I'd say this is good to merge.

**Comment by dce** [comment]
Merged.

**Pull request closed without merging**